bot/cogs/moderation.py

```diff
@@ -181,12 +199,108 @@ async def clear(self, ctx: Context, amount: int) -> None:
+
+    @command()
+    @has_permissions(administrator=True)
+    async def dm(self, ctx: Context, members: Greedy[t.Union[Member, Role]]) -> None:
+        """Dm a List of Specified User from Your Guild."""
+        embed_data = self.embeds_cog.embeds[ctx.author]
+
+        if embed_data.embed.description is None and embed_data.embed.title is None:
+            await ctx.send("Please create a embed using our embed handler to send it.")
+            return
+
+        embed_data.embed.set_footer(text=f"From {ctx.guild.name}", icon_url=ctx.guild.icon_url)
+
+        for member in members:
+            if isinstance(member, Role):
+                for mem in member.members:
+                    await mem.send(embed=embed_data.embed)
+            else:
+                await member.send(embed=embed_data.embed)
+
+    @command()
+    @has_permissions(administrator=True)
+    async def dmall(self, ctx: Context) -> None:
+        """Dm all Users from Your Guild."""
+        embed_data = self.embeds_cog.embeds[ctx.author]
+
+        if embed_data.embed.description is None and embed_data.embed.title is None:
+            await ctx.send("Please create a embed using our embed handler to send it.")
+            return
+
+        embed_data.embed.set_footer(text=f"From {ctx.guild.name}", icon_url=ctx.guild.icon_url)
+
+        for member in ctx.guild.members:
+            with suppress(Forbidden):
+                await member.send(embed=embed_data.embed)
+
+    @command()
+    @has_permissions(manage_channels=True)
+    async def lock(self, ctx, channels: Greedy[TextChannel] = None, reason: str = 'Not Specified') -> None:
+        """Disable @everyone's permission to send message on given channel or current channel if not specified."""
+        if channels is None:
+            channels = [ctx.channel]
+        for channel in channels:
+            await channel.set_permissions(channel.guild.default_role, send_messages=False,
+                                          reason=f"Reason: {reason} | Requested by {ctx.author}.")
+            await channel.send("🔒 Locked down this channel.")
+        if channels != [ctx.channel]:
+            await ctx.send(f"Locked down {len(channels)} channel{'s' if len(channels) > 1 else ''}.")
+
+    @command()
+    @has_permissions(manage_channels=True)
+    async def unlock(self, ctx, channels: Greedy[TextChannel] = None, reason: str = 'Not specified') -> None:
+        """Reset @everyone's permission to send message on given channel or current channel if not specified."""
+        if channels is None:
+            channels = [ctx.channel]
+        for channel in channels:
+            await channel.set_permissions(channel.guild.default_role, send_messages=None,
+                                          reason=f"Reason: {reason} | Requested by {ctx.author}.")
+        await ctx.send(f"Unlocked {len(channels)} channel{'s' if len(channels) > 1 else ''}.")
+
+    @command()
+    @has_permissions(manage_channels=True)
```

Comment:
There might be channels where users with global `manage_channels` permission have this permission explicitly disabled (or vice-versa: users with this permission without having it as global), we should check this permission for every channel they try to use this on instead of checking globally.